internal/metrics/metrics.go

```diff
@@ -90,80 +88,51 @@ func (e *emitter) store() {
 			return
 		}
 
-		identifier := data.Key[0]
 		key := strings.Join(data.Key, ".")
 
-		_, keyExists := e.metricsData[identifier]
-		if !keyExists {
-			e.metricsData[identifier] = make(map[string]float64)
-		}
-
 		//register the metric with go-metrics,
 		//the metric key is used as the identifier.
-		_, registryExists := goMetricsRegistry[identifier]
+		_, registryExists := goMetricsRegistry[key]
 		if !registryExists {
 			switch data.Type {
 			case "counter":
-				c := gometrics.NewCounter()
-				gometrics.Register(key, c)
+				c := gometrics.GetOrRegister(key, gometrics.NewCounter())
 				goMetricsRegistry[key] = c
 			case "gauge":
-				g := gometrics.NewGauge()
-				gometrics.Register(key, g)
+				g := gometrics.GetOrRegister(key, gometrics.NewGauge())
 				goMetricsRegistry[key] = g
 			case "timer":
-				g := gometrics.NewTimer()
-				gometrics.Register(key, g)
-				goMetricsRegistry[key] = g
+				t := gometrics.GetOrRegister(key, gometrics.NewTimer())
+				goMetricsRegistry[key] = t
 			case "histogram":
-				g := gometrics.NewHistogram(gometrics.NewExpDecaySample(1028, 0.015))
-				gometrics.Register(key, g)
-				goMetricsRegistry[key] = g
+				h := gometrics.GetOrRegister(key, gometrics.NewHistogram(gometrics.NewExpDecaySample(1028, 0.015)))
+				goMetricsRegistry[key] = h
 			}
 		}
 
 		//based on the metric type, update the store/registry.
 		switch data.Type {
 		case "counter":
-			e.metricsData[identifier][key] += data.Value
```

Comment:
go-metrics store data in appropriate format, we don't need to do that on our side.